src/gt4sd/algorithms/generation/diffusion/core.py

```diff
@@ -0,0 +1,325 @@
+                algorithm = DiffusersGenerationAlgorithm(configuration=configuration)
+                items = list(algorithm.sample(1))
+                print(items)
+
+            An example for using a conditional generative algorithm from Diffusers::
+
+                configuration = GeneratorConfiguration()
+                algorithm = DiffusersGenerationAlgorithm(configuration=configuration, target="the moon")
+                items = list(algorithm.sample(1))
+                print(items)
+        """
+
+        configuration = self.validate_configuration(configuration)
+        # TODO there might also be a validation/check on the target input
+
+        super().__init__(
+            configuration=configuration,
+            target=target,  # type:ignore
+        )
+
+    def get_generator(
+        self,
+        configuration: AlgorithmConfiguration[S, T],
+        target: Optional[Union[S, T]],
+    ) -> Union[Targeted, Untargeted]:
+        """Get the function to sample batches.
+
+        Args:
+            configuration: helps to set up the application.
+            target: context or condition for the generation.
+
+        Returns:
+            callable generating a batch of items.
+        """
+        logger.info("ensure artifacts for the application are present.")
+        self.local_artifacts = configuration.ensure_artifacts()
+        implementation: Generator = configuration.get_conditional_generator(  # type: ignore
+            self.local_artifacts, target
+        )
+        return implementation.sample
+
+    def validate_configuration(
+        self, configuration: AlgorithmConfiguration
+    ) -> AlgorithmConfiguration:
+        assert isinstance(configuration, AlgorithmConfiguration)
+        return configuration
+
+
+@ApplicationsRegistry.register_algorithm_application(DiffusersGenerationAlgorithm)
+class DiffusersConfiguration(AlgorithmConfiguration[str, None]):
+    """Basic configuration for a diffusion algorithm."""
+
+    algorithm_type: ClassVar[str] = "generation"
+    domain: ClassVar[str] = "vision"
+
+    modality: str = field(
+        default="image",
+        metadata=dict(
+            description="Model domain.  Supported: 'image', 'text', 'audio', 'molecule'."
+        ),
+    )
+
+    model_type: str = field(
+        default="diffusion",
+        metadata=dict(
+            description=f"Type of the model. Supported: {', '.join(MODEL_TYPES.keys())}"
+        ),
+    )
+
+    scheduler_type: str = field(
+        default="discrete",
+        metadata=dict(
+            description=f"Type of the noise scheduler. Supported: {', '.join(SCHEDULER_TYPES.keys())}"
+        ),
+    )
+
+    def get_target_description(self) -> Optional[Dict[str, str]]:
+        """Get description of the target for generation.
+
+        Returns:
+            target description, returns None in case no target is used.
+        """
+        return None
+
+    def get_conditional_generator(
```

Comment:
Here we are breaking the signature by adding target. To keep the implementation consistent with the other algorithms we should remove target and passing it via the configuration since we placed the diffusion algorithm in the generation submodule.
The same pattern is followed from the HuggingFace wrappers.